src/helper/client.cpp

```diff
@@ -24,6 +24,8 @@ namespace dds {
 
 namespace {
 
+class unexpected_command : public std::exception {};
```

Comment:
Add this exception to `exception.hpp` and override what so that we can get some insights if we log it.